torch/_dynamo/variables/nn_module.py

```diff
@@ -67,38 +67,9 @@ def convert_to_fake(x):
         mod._infer_parameters(mod, fake_args, fake_kwargs)
 
 
-def cleanup_source_for_nn_module_stack(source):
-    # TODO(anijain2305, export-team) This is a bad hack to fix the nn module
-    # fully_qualified_name to work with export/unflatten. It converts
-    # mod._modules['net1'] to mod.net1.
-
-    # This type of source occurs when we use UnspecializedNNModule variable
-    # because unspecialized nn module variable inlines module __getattr__ calls.
-    # For export, we rely heavily on NNModuleVariable and do not support
-    # UnspecializedNNModule. But there is one case where this gets exposed -
-    # Pippy. Pippy uses export/unflatten (an export feature) and also
-    # monkepatches the `forward` method of a mod that forces Dynamo to use
-    # UnspecializedNNModule. Therefore, we will need proper work to retain the
-    # nn module stack when we let export rely on UnspecializedNNModule variable.
-
-    # This does not work if we have recursively UnspecializedNNModule variables
-    # e.g. mod._modules['net1']._modules['net2']. This is unlikely to happen in
-    # Pippy so the hotfix is enough for Pippy.
```

Comment:
Its totally ok for Pippy to do that. Its little unusual but Python allows you to do it, so Dynamo should be able to support it. 

As an improvement, It is possible that Pippy could be more functional (don't patch forward). It would be good thing to pursue (but not required).